backend/src/api.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+
+from flask import Blueprint, jsonify, request
+from sqlalchemy.orm.exc import NoResultFound
+
+from .model import Transaction, db
+
+api = Blueprint("api", __name__, url_prefix="/api")
+
+
+@api.route("/transactions/<int:transaction_id>", methods=["PUT", "GET", "DELETE"])
+def transaction(transaction_id):
+    transaction = Transaction.query.filter_by(id=transaction_id).first()
+    if not transaction:
+        return jsonify({"result": "failure", "error": 404}), 404
+
+    if request.method == "PUT":
+        if request.is_json:
+            updates = request.json["updates"]
+            transaction.description = updates["description"]
+            transaction.note = updates["note"]
+            transaction.amount = updates["amount"]
+            db.session.commit()
+            return jsonify(transaction.serialize), 200
+        else:
+            return jsonify({"result": "failure", "error": 400}), 400
```

Comment:
`{"result": "failure", "error": 400}` can be moved to separate function like 'create_error' -> result is always is the same, error code has to be parametrized